client-js/main/client/client.js

```diff
@@ -31,133 +36,141 @@
+   *
+   * @see QueryFactory
+   * @see QueryBuilder
+   * @see AbstractTargetBuilder
    */
-  oneByOne() {
+  newQuery() {
     throw new Error('Not implemented in abstract base.');
   }
 
   /**
-   * Fetches all query results at once fulfilling a promise with an array of entities.
+   * Executes the given `Query` instance specifying the data to be retrieved from
+   * Spine server fulfilling a returned promise with an array of received objects.
    *
-   * @example
-   * // To query all entities of developer-defined Task type at once:
-   * fetchAll({ofType: taskType}).atOnce().then(tasks => { ... })
+   * @param {!Query} query a query instance to be executed
+   * @return {Promise<<T extends Message>[]>} a promise to be fulfilled with a list of Protobuf
+   *        messages of a given type or with an empty list if no entities matching given query
+   *        were found; rejected with a `SpineError` if error occurs;
    *
-   * @return {Promise<Object[]>} a promise to be fulfilled with an array of entities matching query
-   *                             or to be rejected with a `SpineError`
-   * @abstract
+   * @template <T> a Protobuf type of entities being the target of a query
    */
-  atOnce() {
+  execute(query) {
     throw new Error('Not implemented in abstract base.');
   }
-}
-
-/**
- * An abstract client for Spine application backend. This is a single channel for client-server
- * communication in a Spine-based browser application.
- *
- * @abstract
- */
-export class Client {
 
   /**
-   * Defines a fetch query of all entities matching the filters provided as arguments.
-   * This fetch is executed later upon calling the corresponding `.oneByOne()` and
-   * `.atOnce()` methods.
-   *
-   * `fetchAll(...).oneByOne()` queries the entities returning them in asynchronous manner using
-   * an observable. A subscriber is added to an observable to process each next entity or handle
-   * the error during the operation.
-   *
-   * `fetchAll(...).atOnce()` queries all the entities at once fulfilling a returned promise
-   * with an array of objects.
+   * Creates a new {@link TopicFactory} for building subscription topics specifying
+   * the state changes to be observed from Spine server.
    *
    * @example
-   * // Fetch all entities of a developer-defined Task type one-by-one using an observable.
-   * fetchAll({ofType: taskType}).oneByOne().subscribe({
-   *   next(task) { ... },
-   *   error(error) { ... },
-   *   complete() { ... }
-   * })
+   * // Build a subscription topic specifying all entities of a developer-defined `UserTasks` type
+   * newQuery().select(Task)
+   *           .build()
+   *
    * @example
-   * // Fetch all entities of a developer-defined Task type at once using a Promise.
-   * fetchAll({ofType: taskType}).atOnce().then(tasks => { ... })
+   * // Build a subscription topic specifying entities of a developer-defined `UserTasks`
```

Comment:
Also, let's not use the `developer` term. It's a type of the domain entity. So I would say 
```
 Build a subscription topic for `UserTasks` domain entity, selecting the instances which task count is greater than 3.
```